diff --git tensorflow/workspace.bzl tensorflow/workspace.bzl
index dff151246a..f365f557b8 100755
--- tensorflow/workspace.bzl
+++ tensorflow/workspace.bzl
@@ -414,11 +414,11 @@ def tf_workspace(path_prefix = "", tf_repo_name = ""):
 
     tf_http_archive(
         name = "com_github_gflags_gflags",
-        sha256 = "ae27cdbcd6a2f935baa78e4f21f675649271634c092b1be01469440495609d0e",
-        strip_prefix = "gflags-2.2.1",
+        sha256 = "6e16c8bc91b1310a44f3965e616383dbda48f83e8c1eaa2370a215057b00cabe",
+        strip_prefix = "gflags-77592648e3f3be87d6c7123eb81cbad75f9aef5a",
         urls = [
-            "https://mirror.bazel.build/github.com/gflags/gflags/archive/v2.2.1.tar.gz",
-            "https://github.com/gflags/gflags/archive/v2.2.1.tar.gz",
+            "https://mirror.bazel.build/github.com/gflags/gflags/archive/77592648e3f3be87d6c7123eb81cbad75f9aef5a.tar.gz",
+            "https://github.com/gflags/gflags/archive/77592648e3f3be87d6c7123eb81cbad75f9aef5a.tar.gz",
         ],
     )
 
diff --git third_party/tensorrt/BUILD.tpl third_party/tensorrt/BUILD.tpl
index 57682e8735..b1c64f8477 100644
--- third_party/tensorrt/BUILD.tpl
+++ third_party/tensorrt/BUILD.tpl
@@ -34,6 +34,38 @@ cc_library(
     visibility = ["//visibility:public"],
 )
 
+cc_library(
+    name = "nv_infer_plugin",
+    srcs = [%{nv_infer_plugin}],
+    data = [%{nv_infer_plugin}],
+    includes = [
+        "include",
+    ],
+    copts= cuda_default_copts(),
+    deps = [
+        "@local_config_cuda//cuda:cuda",
+        ":tensorrt_headers",
+    ],
+    linkstatic = 1,
+    visibility = ["//visibility:public"],
+)
+
+cc_library(
+    name = "nv_parsers",
+    srcs = [%{nv_parsers}],
+    data = [%{nv_parsers}],
+    includes = [
+        "include",
+    ],
+    copts= cuda_default_copts(),
+    deps = [
+        "@local_config_cuda//cuda:cuda",
+        ":tensorrt_headers",
+    ],
+    linkstatic = 1,
+    visibility = ["//visibility:public"],
+)
+
 
 %{tensorrt_genrules}
 
diff --git third_party/tensorrt/tensorrt_configure.bzl third_party/tensorrt/tensorrt_configure.bzl
index 9b946505a6..61b8f00bff 100644
--- third_party/tensorrt/tensorrt_configure.bzl
+++ third_party/tensorrt/tensorrt_configure.bzl
@@ -19,8 +19,8 @@ load(
 _TENSORRT_INSTALL_PATH = "TENSORRT_INSTALL_PATH"
 _TF_TENSORRT_VERSION = "TF_TENSORRT_VERSION"
 
-_TF_TENSORRT_LIBS = ["nvinfer"]
-_TF_TENSORRT_HEADERS = ["NvInfer.h", "NvUtils.h"]
+_TF_TENSORRT_LIBS = ["nvinfer", "nvinfer_plugin", "nvparsers"]
+_TF_TENSORRT_HEADERS = ["NvInfer.h", "NvUtils.h", "NvInferPlugin.h", "NvUffParser.h", "NvCaffeParser.h"]
 
 _DEFINE_TENSORRT_SONAME_MAJOR = "#define NV_TENSORRT_SONAME_MAJOR"
 _DEFINE_TENSORRT_SONAME_MINOR = "#define NV_TENSORRT_SONAME_MINOR"
